src/compiler/checker.ts

```diff
@@ -6551,19 +6609,30 @@ module ts {
             }
         }
 
+        function createPromiseType(awaitedType: Type, func: FunctionLikeDeclaration): Type {
+            if (globalPromiseSymbol) {
+                awaitedType = getAwaitedType(awaitedType, awaitedType);

```

Comment:
Add a comment for `/* fallbackType */` and a comment specifying why we fall back to the original type.